flocker/node/_deploy.py

```diff
@@ -742,8 +744,10 @@ def calculate_changes(self, desired_configuration, current_cluster_state):
                     for port in application.ports:
                         # XXX: also need to do DNS resolution. See
                         # https://clusterhq.atlassian.net/browse/FLOC-322
-                        desired_proxies.add(Proxy(ip=node.hostname,
-                                                  port=port.external_port))
+                        if node.uuid in node_states:
+                            desired_proxies.add(Proxy(
+                                ip=node_states[node.uuid].hostname,
+                                port=port.external_port))
```

Comment:
Ok. So is this a new bug or is there an existing issue for this?

And if the node subsequently connects to the control service, the proxy will then be configured.

Does the same thing apply to zfs send and  receive state changes?